integration/sawtooth_integration/tests/test_identity.py

```diff
@@ -0,0 +1,399 @@
+# Copyright 2017 Intel Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ----------------------------------------------------------------------------
+
+import enum
+import subprocess
+import unittest
+from uuid import uuid4
+
+import cbor
+
+from sawtooth_processor_test.message_factory import MessageFactory
+
+from sawtooth_integration.tests.integration_tools import RestClient
+from sawtooth_integration.tests.integration_tools import wait_for_rest_apis
+
+from sawtooth_sdk.protobuf import batch_pb2
+
+
+class TestTransactorPermissioning(unittest.TestCase):
+
+    def setUp(self):
+        self.REST_ENDPOINT = "rest-api:8080"
```

Comment:
Only one rest api url is used in this test, right? If so, I would suggest making the url a global constant rather than passing it around as an argument and assigning it as an attribute to all the transactors.

In any case, the `wait_for_rest_apis` call should be separated from the Transactor creation. The usual approach has been to put it in `setUpClass`, since with rest api only has to be waited for once per test.

Reply:
Updating.